### PR TITLE
JavaScript capstone project

### PR DESCRIPTION
**Pull request by banlon-jones**
## JavaScript capstone project
we added 
- comment.js file
- comment.test.js file 
- list items on the homepage using the https://www.tvmaze.com/ API's
- implemented comments using the involvement API provide by microverse

## DEMO link
https://banlon-jones.github.io/JavaScript-API-based-webapp/

## Project video
https://youtu.be/MabsB28cOl8

### CODE REVIEW

File: src/comment.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                        <h2> Comments (${count}) </h2>
+                        <ul class="comments">
+                        </ul>
+                    </div>
+                    <div>
+                        <form id="comment-form" name="comment-form" class="details">
+                            <input type="text" name="name" placeholder="your name" required/>
+                            <input type="text" name="comment" placeholder="Leave a comment" required/>
+                            <button class="btn"> Post </button>
+                        </form>
```
Comment (Meltrust) [inline src/comment.js:34]: - [ ] Upon posting comments, the screen is not updating correctly.  Tested it multiple times and in many instances, you need to make 2 comments for them to appear.  After that, if you make 1 comment it won't appear. 

Please test this carefully, I made a video so you can notice the issue:

https://share.getcloudapp.com/xQuzE6rk

Kindly fix this behavior, so upon posting each comment, the screen updates correctly every time a comment is posted in order to meet this particular requirement:

image



- [ ] OPTIONAL: Upon writing many comments, the popup starts to overflow vertically, and it's a little difficult to keep up the comments form.  You could add a scroll to correct this.

image

If you fix this, you will improve the user experience a lot!!

File: src/involvementAPI.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  });
+};
+
+export const countComments = async (movieID) => {
+  const comments = await getComments(movieID);
+  if (comments.length !== undefined) {
+    return comments.length;
+  }
+  return 0;
+};
```
Comment (Meltrust) [inline src/involvementAPI.js]: Kindly look at this requirement:

image

Therefore, 

- [ ] kindly please make functions to calculate your counters (comments and movies), and please don't use the API to give you those numbers.

- [ ] You will need to update your tests, so they check those functions' correct functionality.

That way we will meet that requirement and get closer to approval 💪

File: src/comment.test.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  },
+));
+
+describe('Testing comment count', () => {
+  document.body.innerHTML = '<div></div>';
+  it('Comments count should be 7', async () => {
+    const count = await countComments();
+    expect(count).toBe(3);
+  });
+});
```
Comment (Meltrust) [inline src/comment.test.js]: Kindly look at this requirement:

image

Therefore, 

- [ ] kindly please make functions to calculate your counters (comments and movies), and please don't use the API to give you those numbers.

- [ ] You will need to update your tests, so they check those functions' correct functionality.

That way we will meet that requirement and get closer to approval 💪

File: src/movieAPI.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+export const getMovie = async (id) => {
+  const response = await fetch(`${baseURL}/shows/${id}`);
+  return response.json();
+};
+
+export const counter = async () => {
+  const movies = await getMovies();
+  return movies.length;
+};
```
Comment (Meltrust) [inline src/movieAPI.js]: Kindly look at this requirement:

image

Therefore, 

- [ ] kindly please make functions to calculate your counters (comments and movies), and please don't use the API to give you those numbers.

- [ ] You will need to update your tests, so they check those functions' correct functionality.

That way we will meet that requirement and get closer to approval 💪

File: src/movieAPI.test.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  },
+));
+
+describe('Testing items count', () => {
+  document.body.innerHTML = '<div></div>';
+  it('item count should be 240', async () => {
+    const count = await counter();
+    expect(count).toBe(240);
+  });
+});
```
Comment (Meltrust) [inline src/movieAPI.test.js]: Kindly look at this requirement:

image

Therefore, 

- [ ] kindly please make functions to calculate your counters (comments and movies), and please don't use the API to give you those numbers.

- [ ] You will need to update your tests, so they check those functions' correct functionality.

That way we will meet that requirement and get closer to approval 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi team, 

### Good job so far!  Your project is beautiful beyond imagination.  It really seems and feels like a movie streaming site!!  Just follow my below recommendations and you will end up with a gorgeous, shiny project ready to wow every crowd!!! 💪 


There are some issues that you still need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/movieAPI.test.js (hunk 5 above)
Comment (banlon-jones) [reply src/movieAPI.test.js]: Hi @Meltrust thanks for the review I have addressed all issues you mentioned thanks a lot

File: src/comment.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  if (!comments.length) {
+    list.innerHTML = '<p>No comments found</p>';
+  } else {
+    comments.forEach((item) => {
+      list.innerHTML += `<li><span>${item.creation_date}</span> | <span>${item.username}:</span> ${item.comment}</li>`;
+    });
+  }
+};
+
+export const countComments = (comments) => {
```
Comment (elisha2kyakpo1) [inline src/comment.js:50]: - [ ] let's make sure that the counter for the comments update after a comment has been added, I had to refresh the page for the comments to update. I'm sure this was working before making the changes.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi team,

Great job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

- I'm happy that all the issues which were mentioned by the previous reviewer has been addressed. Keep it up ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/comment.js (hunk 6 above)
Comment (banlon-jones) [reply src/comment.js:50]: Hi @elisha2kyakpo1, thanks for the feedback, I have fixed the comment counter, all seems good now

**Review by FrederickMih — APPROVED** [review]
#### Hi Team,

## CONGRATULATIONS! :champagne: :champagne: 



In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**